### PR TITLE
Surface API

### LINKED ISSUE #82: Support multiple windows
In Zed, we are initializing a graphics context for every window. However, if there is more than one window, we are wasting time re-initializing the connection to the graphics driver, recompiling all the pipelines, etc. If we support multiple windows natively, we could save that time.

### PR DESCRIPTION
**Pull request by kvark**
Closes #82
TODO:
- [x] API
- [x] Vulkan
- [x] Metal
- [x] GLES
- [ ] Update in Zed
  - I'm confident in the direction here, so testing on Zed can be done in follow-up

Begone the `init_windowed` method. There is only one `init` now. The other one is replaced by `create_surface`.

### CODE REVIEW

**Comment by someone13574** [comment]
I bisected the [BadNativeWindow](https://github.com/kvark/blade/issues/133#issuecomment-2587117350) on gles wayland to this PR using bunnymark. Haven't tested X11 or or my nvidia system yet.

**Comment by kvark** [comment]
Yes, GLES support is a bit more difficult with this new API

File: blade-graphics/src/gles/egl.rs [hunk 1]
```diff
@@ -3,16 +3,17 @@ use std::{
...
 };
 
 const EGL_CONTEXT_FLAGS_KHR: i32 = 0x30FC;
 const EGL_CONTEXT_OPENGL_DEBUG_BIT_KHR: i32 = 0x0001;
-const EGL_PLATFORM_WAYLAND_KHR: u32 = 0x31D8;
-const EGL_PLATFORM_X11_KHR: u32 = 0x31D5;
-const EGL_PLATFORM_XCB_EXT: u32 = 0x31DC;
+const _EGL_PLATFORM_WAYLAND_KHR: u32 = 0x31D8;
+const _EGL_PLATFORM_X11_KHR: u32 = 0x31D5;
+const _EGL_PLATFORM_XCB_EXT: u32 = 0x31DC;
```
Comment (Vanuan) [inline blade-graphics/src/gles/egl.rs]: Is the removal of Wayland / X11 accidental or intentional?

**Comment by Vanuan** [comment]
Is the removal of Wayland / X11 from GLES accidental or intentional?

A bunch of unused variabled in gles/egl:

const _EGL_PLATFORM_WAYLAND_KHR: u32 = 0x31D8;
const _EGL_PLATFORM_X11_KHR: u32 = 0x31D5;
const _EGL_PLATFORM_XCB_EXT: u32 = 0x31DC;

https://github.com/kvark/blade/blob/8cd905be95d9bcf476f69790fcfdbeb615c5154c/blade-graphics/src/gles/egl.rs#L11-L13

**Comment by kvark** [comment]
Implementation for EGL has changed a lot in #310. Also related to #331

**Pull request merged**